location/OBARegionHelper.m

```diff
@@ -90,45 +91,47 @@ - (void)setNearestRegion {
 
         [self.regions removeObjectsInArray:regionsToRemove];
 
-        if (self.regions.count == 0) {
-            [self.appDelegate.modelDao writeSetRegionAutomatically:NO];
-            [self.appDelegate.locationManager removeDelegate:self];
-            [self.appDelegate showRegionListViewController];
-            return;
+        if (self.regions.count != 0) {
+            [self.regions
+             sortUsingComparator:^(id obj1, id obj2) {
+                 OBARegionV2 *region1 = (OBARegionV2 *)obj1;
+                 OBARegionV2 *region2 = (OBARegionV2 *)obj2;
+                 
+                 CLLocationDistance distance1 = [region1 distanceFromLocation:newLocation];
+                 CLLocationDistance distance2 = [region2 distanceFromLocation:newLocation];
+                 
+                 if (distance1 > distance2) {
+                     return (NSComparisonResult)NSOrderedDescending;
+                 }
+                 else if (distance1 < distance2) {
+                     return (NSComparisonResult)NSOrderedAscending;
+                 }
+                 else {
+                     return (NSComparisonResult)NSOrderedSame;
+                 }
+             }];
+            
+            if (self.appDelegate.modelDao.region != nil) {
+                oldRegion = self.appDelegate.modelDao.region.regionName;
+            }
+            
+            [self.appDelegate.modelDao setOBARegion:self.regions[0]];
+            [self.appDelegate refreshSettings];
+            [self.appDelegate.modelDao writeSetRegionAutomatically:YES];
         }
-
-        [self.regions
-         sortUsingComparator:^(id obj1, id obj2) {
-             OBARegionV2 *region1 = (OBARegionV2 *)obj1;
-             OBARegionV2 *region2 = (OBARegionV2 *)obj2;
-
-             CLLocationDistance distance1 = [region1 distanceFromLocation:newLocation];
-             CLLocationDistance distance2 = [region2 distanceFromLocation:newLocation];
-
-             if (distance1 > distance2) {
-                return (NSComparisonResult)NSOrderedDescending;
-             }
-             else if (distance1 < distance2) {
-                return (NSComparisonResult)NSOrderedAscending;
-             }
-             else {
-                return (NSComparisonResult)NSOrderedSame;
-             }
-         }];
-
-        NSString *oldRegion = @"null";
-
+    }
+    
+    [self.appDelegate.locationManager removeDelegate:self];
+    
+    if (self.location == nil || self.regions == nil || self.regions.count == 0){
```

Comment:
also please use `!self.appDelegate.modelDao.region` instead of `self.appDelegate.modelDao.region == nil`